src/coreclr/jit/valuenum.cpp

```diff
@@ -10293,6 +10293,13 @@ void Compiler::fgValueNumberAddExceptionSet(GenTree* tree)
                 fgValueNumberAddExceptionSetForCkFinite(tree);
                 break;
 
+            case GT_XAND:
+            case GT_XORR:
+            case GT_XADD:
+            case GT_XCHG:
+                fgValueNumberAddExceptionSetForIndirection(tree, tree->AsOp()->gtGetOp1());
+                break;
+
 #ifdef FEATURE_HW_INTRINSICS
             case GT_HWINTRINSIC:
                 // ToDo: model the exceptions for Intrinsics
```

Comment:
I am not 100% sure I understand what this `fgValueNumberAddExceptionSet` do, we have a // TODO here for GT_HWINTRINSIC so probably we don't need it for correctness, then it is for optimizations? Do I do the right thing for `GT_XAND\*`?
I checked that adding `fgValueNumberAddExceptionSetForIndirection` for them did not create any diffs so maybe leave them just with break?

Reply:
Thanks, @erozenfeld !

interesting, it means we probably have a preexisting correctness issue for `case GT_HWINTRINSIC:` under that `TODO`. cc @echesakovMSFT, @tannergooding

Reply:
This was added by @briansull when the initial CSE support was being added.